Hello 2 1
2021-06-05 19:47:03.826556: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
2021-06-05 19:47:03.826662: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
WARNING:tensorflow:From /home/parth/.local/lib/python3.8/site-packages/tensorflow/python/compat/v2_compat.py:96: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
Instructions for updating:
non-resource variables are not supported in the long term
Started Reading Flags

checkpoint_load_mode = mru
d_pickle_name = d_processed.p
data_dir = /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC
dropout_keep_prob = 0.8
early_stopping_p = 20
f_d_adam_lr = 0.0003
f_d_batch_size = 16
f_d_class_sampling_str = 10,10,10,10,10,10
f_d_epochs = 7
f_d_metrics_pickle_name = metrics_train_f_on_d.p
f_d_primary_metric = accuracy
f_d_U_adam_lr = 0.0003
f_d_U_batch_size = 32
f_d_U_epochs = 7
f_d_U_metrics_pickle_name = metrics_train_f_on_d_U.p
f_infer_out_pickle_name = infer_f.p
f_layers_str = 512,512
gamma = 0.1
lamda = 0.1
min_rule_coverage = 0
network_dropout = True
num_checkpoints = 1
num_classes = 0
num_load_d = 0
num_load_U = 0
num_load_validation = 0
output_dir = /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3
rule_classes_str = 
run_mode = f_d
shuffle_batches = True
test_w_batch_size = 1000
U_pickle_name = U_processed.p
use_joint_f_w = False
validation_pickle_name = validation_processed.p
w_infer_out_pickle_name = infer_w.p
w_layers_str = 512,512

Hi1
Ended Reading Flags
Loading from hoff  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3/d_processed.p
Loading from hoff  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/U_processed.p
Rule classes:  [2, 1, 1, 4, 3, 2, 1, 3, 5, 4, 3, 2, 4, 1, 2, 5, 4, 1, 2, 4, 3, 1, 3, 2, 1, 1, 4, 4, 5, 1, 3, 5, 2, 1, 5, 1, 5, 5, 0, 5, 5, 1, 4, 2, 3, 2, 1, 3, 5, 1, 1, 3, 2, 1, 1, 5, 2, 3, 2, 5, 3, 3, 4, 5, 5, 4, 2, 0]
length of covered U: 4637
Size of d before oversampling:  68
Size of U (covered) :  4637
Size of d after oversampling:  680
Size of d_U after combining:  5317
Loading 68 elements from d
num instances in d:  680
Loading from hoff  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/validation_processed.p
Setting value of d to 0 for test data
test_w len:  476
test_w len:  476
Number of features:  1024
Number of classes:  6
Print num of rules to train:  68
Print num of rules:  68



Found prev best metric for run type f_d: 0.634
best metrics dict:  {'f1_score_1': 0.6520146520146521, 'precision_1': 0.5933333333333334, 'recall_1': 0.7235772357723578, 'support_1': 123, 'f1_score': array([0.61538462, 0.65201465, 0.49339207, 0.71921182, 0.68253968,
       0.65822785]), 'precision': array([0.8       , 0.59333333, 0.48695652, 0.78494624, 0.82692308,
       0.61176471]), 'recall': array([0.5       , 0.72357724, 0.5       , 0.66363636, 0.58108108,
       0.71232877]), 'support': array([  8, 123, 112, 110,  74,  73]), 'avg_f1_score': 0.6367951151974828, 'avg_precision': 0.6839873124061722, 'avg_recall': 0.6134372412688484, 'accuracy': 0.634, 'epoch': 20.0, 'f_d_global_step': 840}
[Errno 2] No such file or directory: '/home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3/metrics_train_f_on_d_U.p'
Did not find prev best metric for run type f_d_U. Setting to zero.
/home/parth/.local/lib/python3.8/site-packages/tensorflow/python/keras/legacy_tf_layers/core.py:171: UserWarning: `tf.layers.dense` is deprecated and will be removed in a future version. Please use `tf.keras.layers.Dense` instead.
  warnings.warn('`tf.layers.dense` is deprecated and '
/home/parth/.local/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer_v1.py:1719: UserWarning: `layer.apply` is deprecated and will be removed in a future version. Please use `layer.__call__` method instead.
  warnings.warn('`layer.apply` is deprecated and '
WARNING:tensorflow:From /home/parth/.local/lib/python3.8/site-packages/tensorflow/python/util/dispatch.py:201: calling dropout (from tensorflow.python.ops.nn_ops) with keep_prob is deprecated and will be removed in a future version.
Instructions for updating:
Please use `rate` instead of `keep_prob`. Rate should be set to `rate = 1 - keep_prob`.
WARNING:tensorflow:From /home/parth/.local/lib/python3.8/site-packages/tensorflow/python/util/dispatch.py:201: softmax_cross_entropy_with_logits (from tensorflow.python.ops.nn_ops) is deprecated and will be removed in a future version.
Instructions for updating:

Future major versions of TensorFlow will allow gradients to flow
into the labels input on backprop by default.

See `tf.nn.softmax_cross_entropy_with_logits_v2`.

WARNING:tensorflow:From /home/parth/Desktop/SEM6/RnD/spear/spear/Implyloss/my_model.py:615: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.cast` instead.
[
    "<tf.Variable 'global_step:0' shape=() dtype=int64_ref>",
    "<tf.Variable 'f_d_global_step:0' shape=() dtype=int32_ref>",
    "<tf.Variable 'f_network/dense/kernel:0' shape=(1024, 512) dtype=float32_ref>",
    "<tf.Variable 'f_network/dense/bias:0' shape=(512,) dtype=float32_ref>",
    "<tf.Variable 'f_network/dense_1/kernel:0' shape=(512, 512) dtype=float32_ref>",
    "<tf.Variable 'f_network/dense_1/bias:0' shape=(512,) dtype=float32_ref>",
    "<tf.Variable 'f_network/dense_2/kernel:0' shape=(512, 6) dtype=float32_ref>",
    "<tf.Variable 'f_network/dense_2/bias:0' shape=(6,) dtype=float32_ref>",
    "<tf.Variable 'beta1_power:0' shape=() dtype=float32_ref>",
    "<tf.Variable 'beta2_power:0' shape=() dtype=float32_ref>",
    "<tf.Variable 'f_network/dense/kernel/f_d_Adam:0' shape=(1024, 512) dtype=float32_ref>",
    "<tf.Variable 'f_network/dense/kernel/f_d_Adam_1:0' shape=(1024, 512) dtype=float32_ref>",
    "<tf.Variable 'f_network/dense/bias/f_d_Adam:0' shape=(512,) dtype=float32_ref>",
    "<tf.Variable 'f_network/dense/bias/f_d_Adam_1:0' shape=(512,) dtype=float32_ref>",
    "<tf.Variable 'f_network/dense_1/kernel/f_d_Adam:0' shape=(512, 512) dtype=float32_ref>",
    "<tf.Variable 'f_network/dense_1/kernel/f_d_Adam_1:0' shape=(512, 512) dtype=float32_ref>",
    "<tf.Variable 'f_network/dense_1/bias/f_d_Adam:0' shape=(512,) dtype=float32_ref>",
    "<tf.Variable 'f_network/dense_1/bias/f_d_Adam_1:0' shape=(512,) dtype=float32_ref>",
    "<tf.Variable 'f_network/dense_2/kernel/f_d_Adam:0' shape=(512, 6) dtype=float32_ref>",
    "<tf.Variable 'f_network/dense_2/kernel/f_d_Adam_1:0' shape=(512, 6) dtype=float32_ref>",
    "<tf.Variable 'f_network/dense_2/bias/f_d_Adam:0' shape=(6,) dtype=float32_ref>",
    "<tf.Variable 'f_network/dense_2/bias/f_d_Adam_1:0' shape=(6,) dtype=float32_ref>",
    "<tf.Variable 'f_d_U_global_step:0' shape=() dtype=int32_ref>",
    "<tf.Variable 'w_network/w_layer_0/kernel:0' shape=(1092, 512) dtype=float32_ref>",
    "<tf.Variable 'w_network/w_layer_0/bias:0' shape=(512,) dtype=float32_ref>",
    "<tf.Variable 'w_network/w_layer_1/kernel:0' shape=(512, 512) dtype=float32_ref>",
    "<tf.Variable 'w_network/w_layer_1/bias:0' shape=(512,) dtype=float32_ref>",
    "<tf.Variable 'w_network/w_linear_layer/kernel:0' shape=(512, 1) dtype=float32_ref>",
    "<tf.Variable 'w_network/w_linear_layer/bias:0' shape=(1,) dtype=float32_ref>"
]
2021-06-05 19:47:08.807378: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2021-06-05 19:47:08.809051: I tensorflow/compiler/jit/xla_gpu_device.cc:99] Not creating XLA devices, tf_xla_enable_xla_devices not set
2021-06-05 19:47:08.810203: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'libcuda.so.1'; dlerror: libcuda.so.1: cannot open shared object file: No such file or directory
2021-06-05 19:47:08.810236: W tensorflow/stream_executor/cuda/cuda_driver.cc:326] failed call to cuInit: UNKNOWN ERROR (303)
2021-06-05 19:47:08.810267: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:156] kernel driver does not appear to be running on this host (parth-G3-3590): /proc/driver/nvidia/version does not exist
2021-06-05 19:47:08.963001: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:196] None of the MLIR optimization passes are enabled (registered 0 passes)
2021-06-05 19:47:08.970167: I tensorflow/core/platform/profile_utils/cpu_utils.cc:112] CPU Frequency: 2599990000 Hz
Restoring checkpoint from path:  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3/checkpoints/hls-model-2814
Run mode is f_d
training f on d
num_instances:  680
batch_size:  16
total_batch:  42
remaining:  8
total_batch:  42
Optimization started for f_d!
Batch size: 16!
Batches per epoch : 42!
Number of epochs: 7!

========== epoch : 0 ============

cost: 1.3425980563668068e-05

patience: 0


metrics_dict:  {'f1_score_1': 0.6346863468634686, 'precision_1': 0.581081081081081, 'recall_1': 0.6991869918699187, 'support_1': 123, 'f1_score': array([0.615, 0.635, 0.471, 0.718, 0.667, 0.654]), 'precision': array([0.800, 0.581, 0.469, 0.771, 0.808, 0.605]), 'recall': array([0.500, 0.699, 0.473, 0.673, 0.568, 0.712]), 'support': array([  8, 123, 112, 110,  74,  73]), 'avg_f1_score': 0.6267305653803458, 'avg_precision': 0.6722140722616644, 'avg_recall': 0.6041708141670554, 'accuracy': 0.622, 'epoch': 68.0, 'f_d_global_step': 2856}

Reporting f_d metrics to tensorboard
Not saving metrics dict. Best metric value is 0.634 Current is: 0.622

Saved MRU checkpoint to path:  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3/checkpoints/hls-model-2856

No new best checkpoint. Did not save a new best checkpoint. Last checkpointed file:  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3/checkpoints/f_d/best.ckpt-840


========== epoch : 1 ============

cost: 9.922954155800176e-06

patience: 1


metrics_dict:  {'f1_score_1': 0.6327272727272726, 'precision_1': 0.5723684210526315, 'recall_1': 0.7073170731707317, 'support_1': 123, 'f1_score': array([0.615, 0.633, 0.466, 0.718, 0.672, 0.658]), 'precision': array([0.800, 0.572, 0.468, 0.771, 0.824, 0.612]), 'recall': array([0.500, 0.707, 0.464, 0.673, 0.568, 0.712]), 'support': array([  8, 123, 112, 110,  74,  73]), 'avg_f1_score': 0.6271923418598976, 'avg_precision': 0.6744940567502488, 'avg_recall': 0.6040377324790956, 'accuracy': 0.622, 'epoch': 69.0, 'f_d_global_step': 2898}

Reporting f_d metrics to tensorboard
Not saving metrics dict. Best metric value is 0.634 Current is: 0.622

WARNING:tensorflow:From /home/parth/.local/lib/python3.8/site-packages/tensorflow/python/training/saver.py:968: remove_checkpoint (from tensorflow.python.training.checkpoint_management) is deprecated and will be removed in a future version.
Instructions for updating:
Use standard file APIs to delete files with this prefix.
Saved MRU checkpoint to path:  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3/checkpoints/hls-model-2898

No new best checkpoint. Did not save a new best checkpoint. Last checkpointed file:  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3/checkpoints/f_d/best.ckpt-840


========== epoch : 2 ============

cost: 9.276216035557028e-06

patience: 2


metrics_dict:  {'f1_score_1': 0.6402877697841727, 'precision_1': 0.5741935483870968, 'recall_1': 0.7235772357723578, 'support_1': 123, 'f1_score': array([0.615, 0.640, 0.473, 0.718, 0.672, 0.658]), 'precision': array([0.800, 0.574, 0.481, 0.771, 0.824, 0.612]), 'recall': array([0.500, 0.724, 0.464, 0.673, 0.568, 0.712]), 'support': array([  8, 123, 112, 110,  74,  73]), 'avg_f1_score': 0.6295123513231791, 'avg_precision': 0.676967080141495, 'avg_recall': 0.6067477595793668, 'accuracy': 0.626, 'epoch': 70.0, 'f_d_global_step': 2940}

Reporting f_d metrics to tensorboard
Not saving metrics dict. Best metric value is 0.634 Current is: 0.626

Saved MRU checkpoint to path:  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3/checkpoints/hls-model-2940

No new best checkpoint. Did not save a new best checkpoint. Last checkpointed file:  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3/checkpoints/f_d/best.ckpt-840


========== epoch : 3 ============

cost: 8.77108460816176e-06

patience: 3


metrics_dict:  {'f1_score_1': 0.6451612903225806, 'precision_1': 0.5769230769230769, 'recall_1': 0.7317073170731707, 'support_1': 123, 'f1_score': array([0.615, 0.645, 0.475, 0.718, 0.672, 0.658]), 'precision': array([0.800, 0.577, 0.486, 0.771, 0.824, 0.612]), 'recall': array([0.500, 0.732, 0.464, 0.673, 0.568, 0.712]), 'support': array([  8, 123, 112, 110,  74,  73]), 'avg_f1_score': 0.6306843667498446, 'avg_precision': 0.678171972719114, 'avg_recall': 0.6081027731295022, 'accuracy': 0.628, 'epoch': 71.0, 'f_d_global_step': 2982}

Reporting f_d metrics to tensorboard
Not saving metrics dict. Best metric value is 0.634 Current is: 0.628

Saved MRU checkpoint to path:  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3/checkpoints/hls-model-2982

No new best checkpoint. Did not save a new best checkpoint. Last checkpointed file:  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3/checkpoints/f_d/best.ckpt-840


========== epoch : 4 ============

cost: 9.395232820809597e-06

patience: 4


metrics_dict:  {'f1_score_1': 0.6451612903225806, 'precision_1': 0.5769230769230769, 'recall_1': 0.7317073170731707, 'support_1': 123, 'f1_score': array([0.615, 0.645, 0.477, 0.722, 0.672, 0.662]), 'precision': array([0.800, 0.577, 0.491, 0.779, 0.824, 0.609]), 'recall': array([0.500, 0.732, 0.464, 0.673, 0.568, 0.726]), 'support': array([  8, 123, 112, 110,  74,  73]), 'avg_f1_score': 0.6323435575671462, 'avg_precision': 0.6798602161905891, 'avg_recall': 0.6103858781523331, 'accuracy': 0.63, 'epoch': 72.0, 'f_d_global_step': 3024}

Reporting f_d metrics to tensorboard
Not saving metrics dict. Best metric value is 0.634 Current is: 0.63

Saved MRU checkpoint to path:  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3/checkpoints/hls-model-3024

No new best checkpoint. Did not save a new best checkpoint. Last checkpointed file:  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3/checkpoints/f_d/best.ckpt-840


========== epoch : 5 ============

cost: 8.487414509956926e-06

patience: 5


metrics_dict:  {'f1_score_1': 0.6451612903225806, 'precision_1': 0.5769230769230769, 'recall_1': 0.7317073170731707, 'support_1': 123, 'f1_score': array([0.615, 0.645, 0.473, 0.725, 0.672, 0.654]), 'precision': array([0.800, 0.577, 0.481, 0.787, 0.824, 0.605]), 'recall': array([0.500, 0.732, 0.464, 0.673, 0.568, 0.712]), 'support': array([  8, 123, 112, 110,  74,  73]), 'avg_f1_score': 0.6308085708045609, 'avg_precision': 0.6789698625855256, 'avg_recall': 0.6081027731295022, 'accuracy': 0.628, 'epoch': 73.0, 'f_d_global_step': 3066}

Reporting f_d metrics to tensorboard
Not saving metrics dict. Best metric value is 0.634 Current is: 0.628

Saved MRU checkpoint to path:  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3/checkpoints/hls-model-3066

No new best checkpoint. Did not save a new best checkpoint. Last checkpointed file:  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3/checkpoints/f_d/best.ckpt-840


========== epoch : 6 ============

cost: 9.455396710087544e-06

patience: 6


metrics_dict:  {'f1_score_1': 0.6327272727272726, 'precision_1': 0.5723684210526315, 'recall_1': 0.7073170731707317, 'support_1': 123, 'f1_score': array([0.615, 0.633, 0.466, 0.725, 0.672, 0.650]), 'precision': array([0.800, 0.572, 0.468, 0.787, 0.824, 0.598]), 'recall': array([0.500, 0.707, 0.464, 0.673, 0.568, 0.712]), 'support': array([  8, 123, 112, 110,  74,  73]), 'avg_f1_score': 0.626994966199134, 'avg_precision': 0.6748835822107141, 'avg_recall': 0.6040377324790956, 'accuracy': 0.622, 'epoch': 74.0, 'f_d_global_step': 3108}

Reporting f_d metrics to tensorboard
Not saving metrics dict. Best metric value is 0.634 Current is: 0.622

Saved MRU checkpoint to path:  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3/checkpoints/hls-model-3108

No new best checkpoint. Did not save a new best checkpoint. Last checkpointed file:  /home/parth/Desktop/SEM6/RnD/Learning-From-Rules/data/TREC/outputs/f_d_0.1_0.1_3/checkpoints/f_d/best.ckpt-840

Optimization Finished for f_d!
